boxes/generators/angledbox.py

```diff
@@ -108,36 +108,41 @@ def render(self):
 
         with self.saved_context():
             if b != "e":
-                self.floor(x, y , n, edge='f', move="right")
+                self.floor(x, y , n, edge='f', move="right", label="Bottom")
             if self.top == "angled lid":
-                self.floor(x, y, n, edge='e', move="right")
-                self.floor(x, y, n, edge='E', move="right")
+                self.floor(x, y, n, edge='e', move="right", label="Lower Lid")
+                self.floor(x, y, n, edge='E', move="right", label="Upper Lid")
             elif self.top in ("angled hole", "angled lid2"):
-                self.floor(x, y, n, edge='F', move="right", hole=True)
+                self.floor(x, y, n, edge='F', move="right", hole=True, label="Top Rim and Lid")
                 if self.top == "angled lid2":
-                    self.floor(x, y, n, edge='E', move="right")
+                    self.floor(x, y, n, edge='E', move="right", label="Upper Lid")
         self.floor(x, y , n, edge='F', move="up only")
 
         fingers = self.top in ("angled lid2", "angled hole")
 
         self.rectangularWall(lx, h, move="right",
-                             edges=b+"GfG" if fingers else b+"GeG")
+                             edges=b+"GfG" if fingers else b+"GeG",
+                             label="wall {}".format(1))
 
-        for i in range(2):
+        for j in range(2):
             for i in range(n):
                 if i % 2:
                     self.rectangularWall(side, h, move="right",
-                                         edges=b+"GfG" if fingers else b+"GeG")
+                                         edges=b+"GfG" if fingers else b+"GeG",
+                                         label="wall {}".format(i+2+j*n))
                 else:
                     self.rectangularWall(side, h, move="right",
-                                         edges=b+"gfg" if fingers else b+"geg")
+                                         edges=b+"gfg" if fingers else b+"geg",
+                                         label="wall {}".format(i+2+j*n))
 
         if i % 2:
             self.rectangularWall(lx, h, move="right",
-                                 edges=b+"gfg" if fingers else b+"geg")
+                                 edges=b+"gfg" if fingers else b+"geg",
+                                 label="wall {}".format((n+1)*2))
         else:
             self.rectangularWall(lx, h, move="right",
-                                 edges=b+"GfG" if fingers else b+"GeG")
+                                 edges=b+"GfG" if fingers else b+"GeG",
+                                 label="wall {}".format((n+1)*2))
```

Comment:
If the front wall is 1 shouldn't the back wall be n+2  with the second set of walls having larger numbers. May be we should even squeeze the back wall in between the right and the left end pieces...

Reply:
Of course it should. Changed the numbering, but did not rearrange the walls (could not think of a way to do it without duplicating the inner loop or adding ugly branches that makes the logic hard to follow)